### PR TITLE
GH-2056: As a smith I want to migrate the necessary plugin (ui) test suites to LSP-based protocol tests

### PR DESCRIPTION
**Pull request by mor-n4**
See #2056.

### CODE REVIEW

**Comment by mor-n4** [comment]
Note: there are still a few classes with a name matching the pattern `*Plugin*Test`. Most of those are Xpect UI tests. A few others are tests for features that will be removed entirely (e.g. runner UI tests). I did not want to delete them yet, because maybe its better to do this in the context of `IDE-3486`.

**Review by mmews-n4 — APPROVED** [review]
Huge step! This PR reminds me that migrating old code can be harder than writing something completely new!

File: testhelpers/org.eclipse.n4js.ide.tests.helper/src/org/eclipse/n4js/ide/tests/helper/server/AbstractIdeTest.java [hunk 1]
```diff
@@ -829,13 +924,58 @@ protected void renameFileOnDiskWithoutNotification(String moduleName, String new
...
+			if (isFolder) {
+				List<Path> allFiles = FileUtils.getAllRegularFilesIn(newFilePath);
+				for (Path file : allFiles) {
+					fileEvents.add(new FileEvent(toFileURI(file).toString(), FileChangeType.Created));
+				}
+			} else {
+				fileEvents.add(new FileEvent(newFileURI.toString(), FileChangeType.Created));
+			}
+			DidChangeWatchedFilesParams params = new DidChangeWatchedFilesParams(fileEvents);
+			languageServer.didChangeWatchedFiles(params);
```
[the remark this answers is not in the document]
Comment (mor-n4) [reply testhelpers/org.eclipse.n4js.ide.tests.helper/src/org/eclipse/n4js/ide/tests/helper/server/AbstractIdeTest.java:977]: Yes, for now. But we need to revisit this for all methods (as discussed last week).

File: tests/org.eclipse.n4js.ui.tests/src/org/eclipse/n4js/tests/bugs/GHOLD_45_CheckIgnoreAnnotationAtClassLevel_IdeTest.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+import com.google.common.base.Optional;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+
+/**
+ * Test for testing the {@code @Ignore} annotation while running tests via Mangelhaft.
+ */
+// converted from GHOLD_45_CheckIgnoreAnnotationAtClassLevel_PluginUITest
+public class GHOLD_45_CheckIgnoreAnnotationAtClassLevel_IdeTest extends ConvertedIdeTest {
```
[the remark this answers is not in the document]
Comment (mor-n4) [reply tests/org.eclipse.n4js.ui.tests/src/org/eclipse/n4js/tests/bugs/GHOLD_45_CheckIgnoreAnnotationAtClassLevel_IdeTest.java:34]: Exactly. `*IdeTest`s run as ordinary JUnit tests.

File: tests/org.eclipse.n4js.ui.tests/src/org/eclipse/n4js/tests/staticpolyfill/AbstractStaticPolyfillBuilderTest.xtend [hunk 3]
```diff
@@ -11,57 +11,40 @@
...
-import org.eclipse.n4js.tests.util.PackageJSONTestUtils
+import org.eclipse.n4js.projectModel.locations.FileURI
+import org.eclipse.n4js.tests.utils.ConvertedIdeTest
 import org.junit.Before
-import org.eclipse.xtext.resource.SaveOptions
 
 /**
  */
-public abstract class AbstractStaticPolyfillBuilderTest extends AbstractBuilderParticipantTest {
+public abstract class AbstractStaticPolyfillBuilderTest extends ConvertedIdeTest {
```
[the remark this answers is not in the document]
Comment (mor-n4) [reply tests/org.eclipse.n4js.ui.tests/src/org/eclipse/n4js/tests/staticpolyfill/AbstractStaticPolyfillBuilderTest.xtend:22]: Good idea. Renamed.

**Comment by mor-n4** [comment]
Thanks @mmews-n4 !

Builds are green, merging now ...

**Pull request merged**